proposals/0000-swiftpm-binary-dependencies.md

```diff
@@ -181,6 +181,122 @@ while keeping the following as non-goals:
 * Simplicity of binary artifact distribution mechanism
 * Widespread use of binary packages
 
+## New `PackageDescription` API
+
+### BinaryTarget
+Since, a binary target is different compared to a source only target, we propose to introduce a new struct `Artifact`. This struct defines a targets associated artifacts.
+
+```swift
+public struct Artifact {
+    public enum Source {
+        case url(String)
+    }
+
+    public let source: Source
+    public let condition: ArtifactCondition
+}
+```
+
+Furthermore, we propose to add a new `artifacts: [Artifacts]?` property to the `Target`, as well as extend the initlizer with this paramter and create a new static method called `.binaryTarget()`. Lastly, we propose to exten the `TargetType` enum with a new case called `binary`.
+
+
+
+### ArtifactCondition
+To describe for what platform and architecture any given artifact is, we propose to create a new `ArtifactCondition`, similar to the `BuildSettingCondition`.
```

Comment:
I think we should reuse the existing condition expression, but expand it if necessary. I’d like to only end up with one condition expression type.